File: 卡码/代码随想录/1/10-单调栈/Js/LeetCode_0503.js
//503. 下一个更大元素 II
const nextGreaterElements = function(nums) {
	const res=new Array(nums.length).fill(-1);
	const st=new Array();
	for(let i=0;i<nums.length*2;i++){
		while(st.length&&nums[i%nums.length]>nums[st[st.length-1]]){
			res[st[st.length-1]]=nums[i%nums.length]
			st.pop();
		}
		st.push(i%nums.length);
	}
	return res;
};

File: 卡码/代码随想录/1/10-单调栈/Python/LeetCode_0503.py
#503. 下一个更大元素 II
from typing import *

class Solution:
	def nextGreaterElements(self, nums: List[int]) -> List[int]:
		res:List[int]=[-1]*len(nums)
		st:List[int]=[]
		for i in range(len(nums)*2):
			while len(st) and nums[i%len(nums)]>nums[st[-1]]:
				res[st[-1]]=nums[i%len(nums)]
				st.pop()
			st.append(i%len(nums))
		return res